internals/overlord/checkstate/manager.go

```diff
@@ -27,6 +27,7 @@ import (
 // CheckManager starts and manages the health checks.
 type CheckManager struct {
 	mutex           sync.Mutex
+	group           sync.WaitGroup
```

Comment:
Naming nit: this name begs the question, "group of what?" These are often called `wg` by Gophers, or just `waitGroup` if you prefer to spell it out.

Reply:
Changed to ```wg```, thanks.